packages/cosmos/src/osmosis.rs

```diff
@@ -30,6 +32,29 @@ impl Cosmos {
             epochs: res.into_inner().epochs,
         })
     }
+    /// Get the Osmosis txfees information.
+    ///
+    /// Note that this query will fail if called on chains besides Osmosis Mainnet.
+    pub async fn get_osmosis_txfees_info(&self) -> Result<TxFeesInfo, QueryError> {
+        let eip_base_fee = self
+            .perform_query(
+                txfees::QueryEipBaseFeeRequest {},
+                Action::OsmosisTxFeesInfo,
+                true,
+            )
+            .await
+            .map(|res| res.into_inner())?;
+
+        Ok(TxFeesInfo { eip_base_fee })
+    }
+}
+
+/// Information from the txfees module for an Osmosis chain.
+#[derive(Debug)]
+pub struct TxFeesInfo {
+    /// The EIP-1559 base fee, encoded as a string via the Dec type on Osmosis
+    /// this is currently an unsigned integer with 18 decimal places
+    pub eip_base_fee: QueryEipBaseFeeResponse,
```

Comment:
I think it's worth unpacking this into a `Decimal` or similar to avoid needing to deal with decimal point calculations in client code. Basically, the `from_atomics` usage in `cosmos-bin` should move into the library.

Reply:
updated in https://github.com/Levana-Protocol/levana-cosmos-rs/pull/69/commits/788dddb0594498d142cd088485476588b5b0cc52 - had to add some new variants to `Error` and make it `!Clone`, slightly opinionated, but I tried to follow the overall vibe of how error handling is currently done in the lib, lmk if anything should change here